### PR TITLE
Feature/2529 hpo browser exact count

### PR DESCRIPTION
**Pull request by adipaul1981**
image

image

### CODE REVIEW

File: src/components/EntityPage/Participant/ParticipantClinical.js [hunk 1]
```diff
@@ -165,6 +165,14 @@ class ParticipantClinical extends React.Component {
     },
   ];
 
+  filterOutParentDiagnoses(diagnosis) {
+    if ((diagnosis?.mondo?.hits?.edges || []).length > 0) {
```
Comment (evans-g-crsj) [inline src/components/EntityPage/Participant/ParticipantClinical.js:169]: why not putting "diagnosis?.mondo?.hits?.edges" in a const ? E.g: const diagnosisEdges = ...

Comment (evans-g-crsj) [reply src/components/EntityPage/Participant/ParticipantClinical.js:169]: I thinks this class method can be extracted and put "outside" de scope of the class :+1: 


class MyClass {
    method1() {...}
}

VS 

const method1 = () => ....
class MyClass {...}

File: src/components/EntityPage/Participant/ParticipantClinical.js [hunk 2]
```diff
@@ -201,8 +209,10 @@ class ParticipantClinical extends React.Component {
           get(nums[i], 'data.participant.hits.total', '--'),
         );
       }
-
-      this.setState({ diagnoses: sanitize(diagnoses), hasLoadedDxs: true }); //once we're ready, just tell the state, it'll do the rest
+      this.setState({
+        diagnoses: sanitize(diagnoses),
```
Comment (evans-g-crsj) [inline src/components/EntityPage/Participant/ParticipantClinical.js:213]: unneeded space

File: src/components/EntityPage/Participant/ParticipantClinical.js [hunk 3]
```diff
@@ -267,7 +277,7 @@ class ParticipantClinical extends React.Component {
 
   dataIntoState() {
     const api = initializeApi({
```
Comment (evans-g-crsj) [inline src/components/EntityPage/Participant/ParticipantClinical.js:279]: why not importing "initializedApi" (search in the project for an example)

File: src/components/OntologyBrowser/SelectionTree.tsx [hunk 4]
```diff
@@ -55,12 +57,27 @@ export class SelectionTree extends Component<SelectionTreeProps, SelectionTreeSt
...
   ): TreeNode[] =>
     treeNodes
-      .map(({ children, key, title, results, hidden }: TreeNode) => {
+      .map(({ children, key, title, results, hidden, exactTagCount }: TreeNode) => {
         const renderedTitle = (
-          <Fragment>
-            <span>{title}</span>
-            <Tag className="label-document-count">{results}</Tag>
-          </Fragment>
+          <Row justify="space-between" className={'fixed-width-100'}>
```
Comment (evans-g-crsj) [inline src/components/OntologyBrowser/SelectionTree.tsx:62]: if possible, i d extract that into another component <MyTitle props1.../>

File: src/components/OntologyBrowser/SelectionTree.tsx [hunk 5]
```diff
@@ -185,24 +209,40 @@ export class SelectionTree extends Component<SelectionTreeProps, SelectionTreeSt
...
+                type="link"
+                onClick={(e) => {
+                  e.preventDefault();
+                  onItemSelectAll(checkedKeys, false);
+                }}
+              >
+                Clear
+              </Button>
+            </Col>
+            <Col style={{ display: 'flex', alignItems: 'center', paddingRight: 4 }}>
```
Comment (evans-g-crsj) [inline src/components/OntologyBrowser/SelectionTree.tsx:236]: use css vs style

File: src/components/OntologyBrowser/SelectionTree.tsx [hunk 6]
```diff
@@ -185,24 +209,40 @@ export class SelectionTree extends Component<SelectionTreeProps, SelectionTreeSt
...
-              placeholder="Search for ontology term - Min 3 characters"
-              onChange={(e) => this.onChange(e, dataSource)}
-              allowClear
-              size="large"
-            />
+            {selectedField !== 'diagnoses.mondo.name' ? ( //todo remove when mondo search is fixed
+              <Input
+                className="custom-input"
+                placeholder="Search for ontology term - Min 3 characters"
+                onChange={(e) => this.onChange(e, dataSource)}
```
Comment (evans-g-crsj) [inline src/components/OntologyBrowser/SelectionTree.tsx:216]: use class method vs inline

File: src/components/OntologyBrowser/SelectionTree.tsx [hunk 7]
```diff
@@ -185,24 +209,40 @@ export class SelectionTree extends Component<SelectionTreeProps, SelectionTreeSt
...
-              }}
-            >
-              Clear
-            </Button>
+          <Row justify="space-between">
+            <Col>
+              <Button
+                type="link"
+                onClick={(e) => {
+                  e.preventDefault();
```
Comment (evans-g-crsj) [inline src/components/OntologyBrowser/SelectionTree.tsx:229]: use class method vs inline

File: src/components/OntologyBrowser/index.tsx [hunk 8]
```diff
@@ -245,12 +238,17 @@ class OntologyModal extends React.Component<ModalProps, ModalState> {
...
                     targetKeys={targetKeys}
                     onItemSelectAll={onItemSelectAll}
+                    selectedField={this.props.selectedField}
                   />
                 );
               }
             }}
           </Transfer>
         )}
+        <div style={{ color: '#515885' }}>
```
Comment (evans-g-crsj) [inline src/components/OntologyBrowser/index.tsx:248]: use css vs style

File: src/components/OntologyBrowser/store.ts [hunk 9]
```diff
@@ -106,14 +117,18 @@ export class PhenotypeStore {
...
 
-  buildPhenotypeQuery = (field: string, filterThemselves: boolean) => `query($sqon: JSON) {
+  buildPhenotypeQuery = (
+    field: string,
+    filterThemselves: boolean,
+  ) => `query($sqon: JSON, $term_filters: JSON) {
     participant {
       aggregations(filters: $sqon, aggregations_filter_themselves: ${filterThemselves}) {
-        ${field}__name {
+        ${field.replace('.', '__')}__name {
```
Comment (evans-g-crsj) [inline src/components/OntologyBrowser/store.ts:126]: would it make it easier to read if field.replace('.', '__') would be extracted into a function with a meaningful name? const transformStuff = (field) => field.replace('.', '__')

File: src/components/OntologyBrowser/SelectionTree.tsx (hunk 6 above)
Comment (adipaul1981) [reply src/components/OntologyBrowser/SelectionTree.tsx:216]: Je suis un peu moins d'acc avec celle-ci. Techniquement, oui je peux la sortir de la classe mais je pense que ca fait plus de sens de la laisser dans la classe SelectionTree.

**Pull request merged**